src/parcsr_ls/par_amg_setup.c

```diff
@@ -1307,33 +1341,51 @@ hypre_BoomerAMGSetup( void               *amg_vdata,
 
          /**************************************************/
          /*********Set the fixed index to CF_marker*********/
-         //num_C_point_coarse
 
-         if (hypre_ParAMGDataCPointKeepLevel(amg_data) > 0)
+         /* Set fine points (F_PT) given by the user */
+         if ((num_F_points > 0) && (level == 0))
+         {
+            for (j = 0; j < num_F_points; j++)
+            {
+               row = (HYPRE_Int) (F_points_marker[j] - first_local_row);
+               if ((row >= 0) && (row < local_size))
+               {
+                  CF_marker[row] = -1; // Assumes F_PT == -1
+               }
+            }
+         }
+
+         if (hypre_ParAMGDataCPointsKeepLevel(amg_data) > 0)
          {
             if (block_mode)
             {
-               hypre_printf("Keeping coarse nodes in block mode is not implemented\n");
+               if (my_id == 0)
+               {
+                  hypre_printf("Keeping coarse nodes in block mode is not implemented\n");
```

Comment:
We really shouldn't have printf without guarding it print level.  This should probably throw an error code instead.

Reply:
Hi Rob, I modified this to use a call to hypre_error_w_msg instead. See db237b821593a039917c3b91eae42428a0f2d8bc